File: cslacs/Csla/Core/InvalidQueryExpression.cs
using System;
using Csla.Properties;

namespace Csla.Core
{
  class InvalidQueryException : System.Exception
  {
    private string message;

    public InvalidQueryException(string message)
    {
      this.message = message + " ";
    }

    public override string Message
    {
      get
      {
        return Resources.ClientQueryIsInvalid + message;
      }
    }
  }
}

File: cslavb/Csla/Core/InvalidQueryExpression.vb
Namespace Core

  Class InvalidQueryException

    Inherits System.Exception

    Private _messageRenamed As String

    Public Sub New(ByVal message As String)
      Me._messageRenamed = message & " "
    End Sub

    Public Overrides ReadOnly Property Message() As String
      Get
        Return My.Resources.ClientQueryIsInvalid & _messageRenamed
      End Get
    End Property

  End Class

End Namespace